datatypes/src/collections/data_types.rs

```diff
@@ -26,6 +28,25 @@ pub enum VectorDataType {
     MultiPolygon,
 }
 
+impl VectorDataType {
+    pub fn try_from_ogr_type_code(code: u32) -> Result<Self> {
+        Ok(match code {
+            OGRwkbGeometryType::wkbPoint | OGRwkbGeometryType::wkbMultiPoint => {
```

Comment:
if it is called try_from_…, you should implement `TryFrom`

Reply:
TryFrom<u32> doesn't make much sense though.